spec/pry-doc_spec.rb

```diff
@@ -12,6 +12,216 @@
+        expect(described_class.symbol_map).to have_key("foo")
+        expect(described_class.symbol_map).to have_key("baby")
+        expect(described_class.symbol_map).to have_key("wassup")
+        expect(described_class.symbol_map).to have_key("bar")
+        expect(described_class.symbol_map).to have_key("baz")
+        expect(described_class.symbol_map).to have_key("cute_enum_e")
+        expect(described_class.symbol_map).to have_key("baby_enum")
+        expect(described_class.symbol_map).to have_key("cutie_pie")
+      end
+    end
+
+    describe "#fetch_all_definitions" do
+      it "returns both code and file name" do
+        file_ = described_class.symbol_map["foo"].first.file
+        _, file = described_class.new(line_number_style: nil).fetch_all_definitions("foo")
+        expect(file).to eq file_
+      end
+
+      it "returns the code for all symbols" do
+        code, = described_class.new(line_number_style: nil).fetch_all_definitions("foo")
+        expect(decolor code).to include <<EOF
+int
+foo(void) {
+}
+EOF
+
+        expect(decolor code).to include <<EOF
+char
+foo(int*) {
+  return 'a';
+}
+EOF
+      end
+    end
+
+    describe "#fetch_first_definition" do
+      it "returns both code and file name" do
+        code, file = described_class.new(line_number_style: nil).fetch_first_definition("wassup")
+        expect(decolor code).to include "typedef int wassup;"
+        expect(file).to eq File.join(__dir__, "fixtures/c_source/hello.c")
+      end
+
+      context "with line numbers" do
+        context "normal style (actual line numbers)" do
+          it "displays actual line numbers" do
+            code, = described_class.new(line_number_style: :'line-numbers').fetch_first_definition("bar")
+            expect(decolor code).to include <<EOF
+11: enum bar {
+12:   alpha,
+13:   beta,
+14:   gamma
+15: };
+EOF
+          end
+
+          context "base one style (line numbers start with 1)" do
+            it "displays actual line numbers" do
+              code, = described_class.new(line_number_style: :'base-one').fetch_first_definition("bar")
+              expect(decolor code).to include <<EOF
+1: enum bar {
+2:   alpha,
+3:   beta,
+4:   gamma
+5: };
+EOF
+            end
+          end
+        end
+      end
+
+      it "returns the code for a function" do
+        code, = described_class.new(line_number_style: nil).fetch_first_definition("foo")
+        expect(decolor code).to include(<<EOF
+int
+foo(void) {
+}
+EOF
+                                       ).or include <<EOF
+char
+foo(int*) {
+  return 'a';
+}
+EOF
+      end
+
+      it "returns the code for an enum" do
+        code, = described_class.new(line_number_style: nil).fetch_first_definition("bar")
+        expect(decolor code).to include <<EOF
+enum bar {
+  alpha,
+  beta,
+  gamma
+};
+EOF
+      end
+
+      it "returns the code for a macro" do
+        code, = described_class.new(line_number_style: nil).fetch_first_definition("baby")
```

Comment:
If `line_number_style: nil` is assumed by default, we probably don't need to set it to `nil` explicitly every time. What do you think?